File: backend/src/main/java/com/backend/backend/dao/entities/FuiteMessage.java
package com.backend.backend.dao.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.Date;

@Entity
@Table(name = "fuite_messages")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FuiteMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long utilisateurId;

    @Lob
    @Column(columnDefinition = "TEXT")
    private String contenuTexte;

    private String cheminAudio;

    private Integer dureeAudioSecondes;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(nullable = false)
    private Date dateEnvoi;

    @ManyToOne
    @JoinColumn(name = "fuite_id", nullable = false)
    private Fuite fuite;
}

File: frontend/lib/models/fuite_message.dart
class FuiteMessage {
  final int id;
  final int utilisateurId;
  final String? nomUtilisateur;
  final String? contenuTexte;
  final String? cheminAudio;
  final int? dureeAudioSecondes;
  final String? dateEnvoi;
  final int fuiteId;

  FuiteMessage({
    required this.id,
    required this.utilisateurId,
    this.nomUtilisateur,
    this.contenuTexte,
    this.cheminAudio,
    this.dureeAudioSecondes,
    this.dateEnvoi,
    required this.fuiteId,
  });

  factory FuiteMessage.fromJson(Map<String, dynamic> json) {
    return FuiteMessage(
      id: json['id'] as int,
      utilisateurId: json['utilisateurId'] as int,
      nomUtilisateur: json['nomUtilisateur'] as String?,
      contenuTexte: json['contenuTexte'] as String?,
      cheminAudio: json['cheminAudio'] as String?,
      dureeAudioSecondes: json['dureeAudioSecondes'] as int?,
      dateEnvoi: json['dateEnvoi'] as String?,
      fuiteId: json['fuiteId'] as int,
    );
  }
}
